### PR TITLE
Use faster Guid/Name lookup for EventSource

### PR DESCRIPTION
**Pull request by benaadams**
If we already have the `EventSource` instance we can just get the name and guid from that, rather than looking them up using reflection on the type

/cc @tommcdon @krwq @eerhardt

### CODE REVIEW

**Comment by tommcdon** [comment]
@sywhang @josalem

**Review by sywhang — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]

**Review by josalem — APPROVED** [review]

**Comment by benaadams** [comment]
Btw; while this has 3 approvals, I don't have commit rights so can't merge it 😉

**Comment by sywhang** [comment]
Just merged it : ) Thanks @benaadams !

**Pull request merged**
